salvador/fetchImage.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  fetchImage.swift
+//  salvador
+//
+//  Created by Geury Roustand on 06.02.23.
+//
+
+import SwiftUI
+
+enum ImageError: Error {
+    case badURL
+}
+
+
+class createImage {
+    
+    static let shared = createImage()
+    
+    func generateImage(withPrompt prompt: String, apiKey: String) async throws -> apiData {
+        
+        guard let url = URL(string: "https://api.openai.com/v1/images/generations") else {
+            throw ImageError.badURL
+        }
+  
+        // Dicionary  
+        let parameters: [String: Any] = [
+            "prompt":prompt,
+            "n": 1,
+            "size":"1024x1024"
+        ]
+        
+        let data: Data = try JSONSerialization.data(withJSONObject: parameters)
```

Comment:
You could use `JSONEncoder` instead of `JSONSerialization` like you used `JSONDecoder` already below. `JSONSerialization` is an older API from objective-C times.